catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -109,6 +133,75 @@ public void init() {
           res.header("Content-Encoding", "gzip");
         });
 
+    post(
+        "/cql/transform/:transformerId",
+        (req, res) -> {
+          String transformerId = req.params(":transformerId");
+          String body = util.safeGetBody(req);
+          CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+          executeCqlQuery(cqlRequest);
+
+          for (ServiceReference<QueryResponseTransformer> queryResponseTransformer :
+              queryResponseTransformers) {
+
+            LOGGER.trace("Finding transformer to transform query response.");
+
+            String id = (String) queryResponseTransformer.getProperty("id");
+
+            if (id.equals(transformerId)) {
+              BinaryContent content =
+                  bundleContext
+                      .getService(queryResponseTransformer)
+                      .transform(this.queryResponse, new HashMap<>());
+
+              String mimeType = (String) queryResponseTransformer.getProperty("mime-type");
+              MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+              String fileExt = allTypes.forName(mimeType).getExtension();
+
+              res.status(200);
+
+              String acceptEncoding = req.headers("Accept-Encoding");
```

Comment:
[HttpHeaders](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/HttpHeaders.html) has constants for these.